include/counter.h

```diff
@@ -97,7 +97,9 @@ static inline u32_t counter_read(struct device *dev)
  * @param dev Pointer to the device structure for the driver instance.
  * @param callback Pointer to the callback function. if this is NULL,
  *                 this function unsets the alarm.
- * @param count Number of counter ticks.
+ * @param count Number of counter ticks. This is relative value, it means that
```

Comment:
Slight tweak:

     @param count Number of counter ticks.  This is a relative value, meaning
                  an alarm will be triggered count ticks in the future.